packages/govuk-frontend/src/govuk/components/header/header.mjs

```diff
@@ -83,23 +83,45 @@ export class Header extends GOVUKFrontendComponent {
     this.$menu = $menu
     this.$menuButton = $menuButton
 
-    // Set the matchMedia to the govuk-frontend desktop breakpoint
-    this.mql = window.matchMedia('(min-width: 48.0625em)')
+    this.setupResponsiveChecks()
+
+    this.$menuButton.addEventListener('click', () =>
+      this.handleMenuButtonClick()
+    )
+  }
+
+  /**
+   * Setup viewport resize check
+   *
+   * @private
+   */
+  setupResponsiveChecks() {
+    const breakpointProperty = '--govuk-frontend-breakpoint-desktop'
+    const breakpointValue = getComputedStyle(document.body).getPropertyValue(
+      breakpointProperty
+    )
```

Comment:
**question** How would you feel about encapsulating that into a `getBreakpointValue(breakpointName)`. Both to wrap what's needed to access the value out of each class and to see if services maybe start picking up the pattern by digging into our private code?

Reply:
Absolutely fine by me

Reply:
I've added `getBreakpoint()` in bf557114a4a40c2afd5c150f986297bcd5050ae2 as we'll still need the CSS property name for the error

```mjs
const breakpoint = getBreakpoint('tablet')

// breakpoint.property
// breakpoint.value
```